src/CFamily/CMake/CompilationDatabaseLocator.cs

```diff
@@ -52,68 +55,68 @@ internal class CompilationDatabaseLocator : ICompilationDatabaseLocator
 
         [ImportingConstructor]
         public CompilationDatabaseLocator(IFolderWorkspaceService folderWorkspaceService, ILogger logger)
-            : this(folderWorkspaceService, new FileSystem(), new BuildConfigProvider(logger), logger)
+            : this(folderWorkspaceService,
+                new BuildConfigProvider(logger),
+                new FileSystem(),
+                logger)
         {
         }
 
-        public CompilationDatabaseLocator(IFolderWorkspaceService folderWorkspaceService, IFileSystem fileSystem,
-            IBuildConfigProvider activeConfigProvider, ILogger logger)
+        public CompilationDatabaseLocator(IFolderWorkspaceService folderWorkspaceService, 
+            IBuildConfigProvider buildConfigProvider,
+            IFileSystem fileSystem, 
+            ILogger logger)
         {
             this.folderWorkspaceService = folderWorkspaceService;
             this.fileSystem = fileSystem;
-            this.buildConfigProvider = activeConfigProvider;
+            this.buildConfigProvider = buildConfigProvider;
             this.logger = logger;
         }
 
         public string Locate()
```

Comment:
Previously the code would call GetCompilationDatabaseFolderPath and then check if there is a file under that folder. 
But in the end it would return only the file path, as the deserialization is done in the new `CompilationConfigProvider`.
So it looked awkward to test for file existence here.
I removed it and now essentially the logic is only the previous GetCompilationDatabaseFolderPath.